src/utils/ParseRouteUtils.js

```diff
@@ -512,14 +587,53 @@ function parseBusRoutes(busRoutes: Array<Object>, destinationName: string): Prom
         };
       }));
 
+      const startName = directions.length > 0 ? directions[0].name : null;
+      const endName = directions.length > 0 ? directions[directions.length - 1].name : null;
+
+      let previousStopName;
+      const routeSummary = directions.map((dir, index) => {
+        const { name, type, stayOnBusForTransfer } = dir;
+        // We want to ignore the intial walking direction
+        if (index === 0 && type === DIRECTION_TYPE.WALK) {
+          return null;
+        }
+        const direction = {
+          busNumber: null,
+          type,
+        };
+        let stopName = name;
+        if (type === DIRECTION_TYPE.DEPART) {
+          direction.busNumber = dir.routeNumber;
+          previousStopName = dir.stops[dir.stops.length - 1].name;
+        } else {
+          // If the direction type is walking, the stopName should be the name of the stop we just got off of.
```

Comment:
prefer this comment on the previous line:

`If walking direction type, stopName should be the name of the stop we just got off`